### DISCUSSION TITLE
Need help in forwarding a stream from one network to another

### DISCUSSION THREAD

**Discussion by sndkonstantin** [bluenviron/mediamtx#660] [Questions]
Hello! I am pretty new to RTSP and all other web technologies. I want to get the RTSP stream from the IP camera from the network without STATIC IP (SOURCE) and pass it to my server with STATIC IP (SERVER). The way I am doing it at the moment does not seem right to me. I use OBS to pass RTMP stream with configurations rtmp://DOMAINNAME:6667 . The port :6667 becomes :1935 and the instance of rtsp-simple-server works great and gives me the desired RTSP stream at rtsp://localhost:8554/path.  But I seems to me that the proper way was to use something like this on SOURCE PC :

code code code
#####
paths:
  redirected:
    source: rtsp://user:password@192.168.50.15:554/ #### THIS IS MY IP CAM
    sourceRedirect: rtsp://user:password@DOMAINNAME:8554/path ##### THIS IS WHERE I WANT STREAM TO COME 

but there is no stream on SERVER.  port :8554 is forwarded to :8554 
Both systems are WIN 10. Maybe I miss something. I will be grateful for any piece of advice.

**Comment by aler9** [comment]
Hello, as explained in the README and in the configuration file, `sourceRedirect` is not used to proxy a stream from a server to another, but has another meaning.
To proxy a stream from a camera to a remote server, you can simply use ffmpeg, you don't need a local server:

```
ffmpeg -re -i rtsp://user:password@192.168.50.15:554/ -c copy -f rtsp rtsp://user:password@DOMAINNAME:8554/path
```

**Reply by migvel** [reply]
Hello, I'm trying to do the same but with RTSPS, I cannot make ffmpeg to work using that example with TLS.

I have a local server and I want to stream to a remote server with RTSPS, do you know how to do that? I use:
`ffmpeg -re -i rtsps://userlocal:passlocal@url:port/origin -c copy -f rtsps rtsps:/ /userremote:passremote@urlremote:port/destiny`

But I get: Requested output format 'rtsps' is not suitable output format.

Thanks.

**Comment by migvel** [comment]
I was using wrongly TLS  with FFMPEG, after reading this document: http://underpop.online.fr/f/ffmpeg/help/tls.htm.gz

This is the suggested way, just for streaming with TLS:
ffmpeg -i input -f rtsp tls://hostname:port?listen&cert=server.crt&key=server.key

So I do that but I get an error in the rtsp-simple-server side: 
ERR: tls: first record does not look like a TLS handshake 

Any help would be very welcome.

**Comment by migvel** [comment]
Finally I got it to work,

 ffmpeg -re -i "rtsps://usr:pass@localhost:port/local?listen&cert=server.crt&key=server.key" -c copy -f rtsp "rtsps://usr:pass@localhost:port/remote?listen&cert=server.crt&key=server.key"

I didn't check yet if this is properly encrypted.

**Reply by aler9** [reply]
you can skip the prefix `?listen&cert=server.crt&key=server.key` since it isn't needed. ffmpeg doesn't perform certificate validation.

[a comment by an automated account is not shown]
